packages/core-sdk/src/utils/registerHelper.ts

```diff
@@ -0,0 +1,1319 @@
+    [SignatureMethodType.REGISTER]: {
+      ...baseConfig,
+      permissions: [
+        {
+          ipId: ipIdAddress,
+          signer: registrationWorkflowsAddr,
+          to: coreMetadataModuleAddr,
+          permission: AccessPermission.ALLOW,
+          func: getFunctionSignature(coreMetadataModuleAbi, "setAll"),
+        },
+      ],
+    },
+    [SignatureMethodType.BATCH_REGISTER_DERIVATIVE]: {
+      ...baseConfig,
+      state: state!,
+      to: licensingModuleAddr,
+      encodeData: encodeData!,
+      verifyingContract: ipIdAddress,
+    },
+  };
+  const signatureRequest = signatureConfigs[methodType];
+  if (
+    (methodType === SignatureMethodType.DISTRIBUTE_ROYALTY_TOKENS ||
+      methodType === SignatureMethodType.BATCH_REGISTER_DERIVATIVE) &&
+    signatureRequest
+  ) {
+    const { signature } = await getSignature(signatureRequest as SignatureRequest);
+    return signature;
+  } else {
+    const { signature } = await getPermissionSignature(
+      signatureRequest as PermissionSignatureRequest,
+    );
+    return signature;
+  }
+};
+
+export const handleDistributeRoyaltyTokensRequest = async <
+  T extends TransformIpRegistrationWorkflowRequest,
+>({
+  request,
+  rpcClient,
+  wallet,
+  chainId,
+}: HandleDistributeRoyaltyTokensRequestConfig): Promise<
+  TransformIpRegistrationWorkflowResponse<T>
+> => {
+  const { ipId, deadline, ipRoyaltyVault, totalAmount } = request;
+  const calculatedDeadline = await getCalculatedDeadline(rpcClient, deadline);
+  const ipRoyaltyVaultImpl = new IpRoyaltyVaultImplReadOnlyClient(rpcClient, ipRoyaltyVault);
+  const balance = await ipRoyaltyVaultImpl.balanceOf({ account: ipId });
+  const royaltyTokenDistributionWorkflowsClient = new RoyaltyTokenDistributionWorkflowsClient(
+    rpcClient,
+    wallet,
+  );
+  if (BigInt(balance) < BigInt(totalAmount)) {
+    throw new Error(
+      `The balance of the IP account in the IP Royalty Vault is insufficient to distribute the royalty tokens.`,
+    );
+  }
+  const ipAccount = new IpAccountImplClient(rpcClient, wallet, validateAddress(ipId));
+  const { result: state } = await ipAccount.state();
+  const signatureApproveRoyaltyTokens = await generateOperationSignature({
+    ipIdAddress: ipId,
+    methodType: SignatureMethodType.DISTRIBUTE_ROYALTY_TOKENS,
+    deadline: calculatedDeadline,
+    ipRoyaltyVault: ipRoyaltyVault,
+    totalAmount: totalAmount,
+    state,
+    encodeData: encodeFunctionData({
+      abi: ipRoyaltyVaultImplAbi,
+      functionName: "approve",
+      args: [royaltyTokenDistributionWorkflowsAddress[chainId], BigInt(totalAmount)],
+    }),
+    wallet: wallet,
+    chainId: chainId,
+  });
+  const transformRequest = {
+    ipId,
+    royaltyShares: request.royaltyShares,
+    sigApproveRoyaltyTokens: {
+      signer: wallet.account!.address,
+      deadline: calculatedDeadline,
+      signature: signatureApproveRoyaltyTokens,
+    },
+  };
+  return {
+    transformRequest: transformRequest as T,
```

Comment:
casting really defeat the purchase of all the typing work that we are doing. We should avoid casting as much as possible but some casts are necessary so I suggest adding a comment above every cast used indicating why its absolutely needed.